### PR TITLE
feat: disable automation rules if condition fails multiple times

### PR DESCRIPTION
**Pull request by scmmishra**
This PR adds a feature to disable automation rules if the conditions fail multiple times. It also sends an email to the administrators notifying them of this change.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Unit Tests
- [ ] Manual Test on Staging

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by scmmishra** [comment]
@sojan-official PTAL whenever you can

File: app/models/automation_rule.rb [hunk 1]
```diff
@@ -20,9 +20,13 @@
 class AutomationRule < ApplicationRecord
   include Rails.application.routes.url_helpers
 
+  MAX_ERROR_THRESHOLD = 3
```
Comment (sojan-official) [inline app/models/automation_rule.rb:23]: can't we use reauthorizable concern for this case as well ?  seems like lot of logic is pretty common

**Review by sojan-official — COMMENTED** [review]
@scmmishra  I think we can handle this via `reauthorizable` concern. we wouldn't need to include a lot of extra code.

File: app/models/automation_rule.rb (hunk 1 above)
Comment (scmmishra) [reply app/models/automation_rule.rb:24]: @sojan-official I've made the changes, let me test it out properly. I'll let you know once it's ready for review

File: app/models/concerns/error_trackable.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def threshold_breached?
+    error_count >= error_threshold
+  end
+
+  private
+
+  def error_count_key
+    format(::Redis::Alfred::ERROR_TRACKABLE_COUNT, obj_type: self.class.table_name.singularize, obj_id: id)
```
Comment (scmmishra) [inline app/models/concerns/error_trackable.rb:34]: This PR changes the key to a standard one across the app, the Reauthorizable had a specific key. Is this fine? @sojan-official

**Review by sojan-official — APPROVED** [review]
@scmmishra made some slight changes to the code so that we can include the shared reauthorizable specs into the automation rule model specs as well. 

Please check if i missed something

**Pull request merged**